### DISCUSSION TITLE
Question: How to control the colors of new custom completion menu?

### DISCUSSION THREAD

**Discussion by poetaman** [neoclide/coc.nvim#4035] [Q&A]
Am unable to change the colors of all the fields of the custom completion menu. Here's what I need:

* Highlight group that can be used to change color of completion-type field [LS] [S] [A]
* Unlink highlight of search characters to CocSearch
* Most importantly: apply reverse for foreground and background of completion menu, but don't completion-symbol, completion-type, and completion-search characters reverse! This is essential for people coding a base16 color scheme that relies on cterm colors. Relying on cterm colors has the benefit that if underlying terminal colors are changed, neovim automatically adapts to the change. The most sensible color choice for completion menu is to reverse the foreground and background, that way it would work for all color schemes whether dark or light.

<img width="199" alt="Screen Shot 2022-08-09 at 4 38 19 PM" src="https://user-images.githubusercontent.com/71736629/183780751-1fe4ce1a-122a-46cc-bc4e-872c1ac5966c.png">

**Comment by chemzqm** [comment]
It's a custom float window/popup with other highlights, so I don't think it's a good idea with reversed highlight, create custom highlight to `CocFloating` is recommended and your can use `border` in `floatConfig`

**Comment by poetaman** [comment]
@chemzqm Ok thanks. How do I change color of the completion type in above image? [LS], [S], [A]??

**Reply by chemzqm** [reply]
Create your custom highlight of `CocFloating` or use `"suggest.floatConfig"` with custom highlight group.

**Comment by poetaman** [comment]
Am unable to change color of the selected item in float. How to set that? In my simple test-case I am trying to use only colors 0 (black) and 7 (white) for background and foreground of completion selection

```
hi! Pmenu ctermfg=0 ctermbg=7
hi! PmenuSel ctermfg=7 ctermbg=0
hi! CocPumMenu ctermfg=0 ctermbg=7
hi! CocPumSearch ctermfg=NONE ctermbg=NONE cterm=underline
```

It leads to the following:

<img width="426" alt="Screen Shot 2022-08-09 at 6 36 52 PM" src="https://user-images.githubusercontent.com/71736629/183792547-b5e98ba7-be85-45e1-8487-2f7fe109f856.png">

I would like to fix 2 things:
* The selected item should show ctermfg=7 and ctermbg=0, which it isn't. 
* Change the color of the preview window that opens to also be just black and white. Not sure yet what color groups it uses for item like fu[nction] which seems to use cyan.

**Reply by chemzqm** [reply]
Check out `:h coc-completion`

**Reply by poetaman** [reply]
Ok, I was able to correct the highlight ``CocMenuSel `` of selected item (as seen below). Though I was not able to find information on few highlight groups in the image:
* The cyan "fu[nction]" in the heading of the preview box of above image.
* How to disable syntax highlighting in the preview box for snippets of below image.
* How to disable color of completion-type?? [S] [LS] [A] [B]??

<img width="464" alt="Screen Shot 2022-08-09 at 9 10 54 PM" src="https://user-images.githubusercontent.com/71736629/183814234-2c99978e-c302-4e25-bc8d-bda548b1d64c.png">

```
hi! Pmenu ctermfg=0 ctermbg=7
hi! PmenuSel ctermfg=7 ctermbg=0
hi! CocPumMenu ctermfg=0 ctermbg=7
hi! CocMenuSel ctermfg=7 ctermbg=0
hi! CocPumSearch ctermfg=NONE ctermbg=NONE cterm=underline
```

As a side note it would help you to maintain a set of screenshots that show which highlight group gets applied where.

**Reply by graue** [reply]
@poetaman, thanks for this! The menu was showing up in unreadable colors for me, and putting in your color settings made it look nice.
